### ISSUE TITLE
Feature: Pinned-Dependencies should ignore Dockerfiles and scripts in test folders

### ISSUE THREAD

**Issue by pnacht** [ossf/scorecard#3315]
**Is your feature request related to a problem? Please describe.**
Pinned-Dependencies currently complains about Dockerfiles (and other scripts, I assume?) that are clearly marked as being used for tests.

**Describe the solution you'd like**
Similarly to Binary-Artifacts, which excuses test binaries, Pinned-Dependencies should excuse clearly identified test dependencies.

For example, [aomediacodec/libavif](https://github.com/aomediacodec/libavif) is currently penalized for having an unpinned Dockerfile in `./tests/docker/Dockerfile`.

**Comment by spencerschrock** [comment]
@gabibguti this is something maintainer annotations could fix, 

No action needed right now, but perhaps a low priority feature once the design lands would be declaring a test folder, instead of declaring on a per-probe/check basis.
